### PR TITLE
Enhance capataz error reporting to be human-friendly

### LINKED ISSUE #61: Enhance error reporting
### Context

Currently, we rely on the notification system to submit errors to third-party error trackers. We have discovered this approach is boisterous for every error in the system that is recoverable.

We also have discovered that the metadata approach we attempted to get details on the error messages is not working well. It is hard to see the story of a given error, and some third-party error trackers limit the number of labels an error can have.

### Acceptance Criteria

* Ensure errors have a reported string with details what went wrong and why in a human-friendly way

### PR DESCRIPTION
**Pull request by roman**
Closes #61

### CODE REVIEW

File: cap/error_exports.go [hunk 1]
```diff
@@ -41,3 +41,9 @@ type SupervisorRestartError = s.SupervisorRestartError
 //
 // Since: 0.0.0
 type RestartToleranceReached = s.RestartToleranceReached
+
+// ExplainError is an utility function to get a human-friendly message from a
```
Comment (julio-garcia-fc) [inline cap/error_exports.go:45]: nit: s/is an/is a/

File: internal/s/error.go [hunk 2]
```diff
@@ -71,6 +77,142 @@ func (err *SupervisorTerminationError) KVs() map[string]interface{} {
...
 }
 
+// explainLines returns a human-friendly message of the error represented as a slice
+// of lines
+func (err *SupervisorTerminationError) explainLines() []string {
+	// error reporting from children
+	var nodeErrLines []string
+
+	// deal with a single child error
+	if err.nodeErrMap != nil && len(err.nodeErrMap) == 1 {
```
Comment (julio-garcia-fc) [inline internal/s/error.go:87]: You do not really need `err.nodeErrMap != nil`. I am pretty sure `len()` handles that for you.

**Review by julio-garcia-fc — COMMENTED** [review]
Where do you anticipate this will get captured? Is it structured logging friendly? The messages seem appropriate for a console. Is that the intent?

I do not have a good mental model for the impact on future functionality enticements? 

Also, for the future, a key-value tree may be more general purpose for I18N or other analysis (e.g. problem children nodes, common failures, ...).

I am curious about your thoughts?

The messages are awesome!!!

**Comment by roman** [comment]
> Where do you anticipate this will get captured? Is it structured logging friendly? The messages seem appropriate for a console. Is that the intent?

The idea is to use this report error as the error message that our third-party error system receives.

> Also, for the future, a key-value tree may be a more general-purpose for I18N or other analysis (e.g. problem children nodes, common failures, ...).

The KV tags supposedly allow this kind of analysis, although a bit unstructured. Dealing with error analysis is not easy if a well-defined abstract contract is present, and to be honest, I'm learning here. We will figure it out as we go.

**Review by julio-garcia-fc — APPROVED** [review]

**Pull request merged**